context/jfr/src/main/java/brave/context/jfr/JfrScopeDecorator.java

```diff
@@ -0,0 +1,70 @@
+package brave.context.jfr;
+
+import brave.internal.HexCodec;
+import brave.internal.Nullable;
+import brave.propagation.CurrentTraceContext.Scope;
+import brave.propagation.CurrentTraceContext.ScopeDecorator;
+import brave.propagation.TraceContext;
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Event;
+import jdk.jfr.Label;
+
+/**
+ * Adds {@linkplain Event} properties "traceId", "parentId" and "spanId" when a {@link
+ * brave.Tracer#currentSpan() span is current}. These can be used to correlate JDK Flight recorder
+ * events with logs or Zipkin data correlation.
+ *
+ * <p>Ex.
+ * <pre>{@code
+ * tracing = Tracing.newBuilder()
+ *                  .currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
+ *                    .addScopeDecorator(JfrScopeDecorator.create())
+ *                    .build()
+ *                  )
+ *                  ...
+ *                  .build();
+ * }</pre>
+ */
+public final class JfrScopeDecorator implements ScopeDecorator {
+
+  @Category("Zipkin")
```

Comment:
super cool that from here down is less than 50loc. it is very easy to implement this, almost copy/pasteable!